services/concept_insights/ci.js

```diff
@@ -0,0 +1,422 @@
+          //Check fields are populated
+          if (document_name && document_label && content_type) {
+            params = {
+              'id': '/corpora/'+account_id+'/'+corpus_name+'/documents/'+document_name,
+              'document': {
+                'label': document_label,
+                'parts': [
+                {
+                  'name': 'part-one',
+                  'content-type': content_type,
+                  'data': msg.payload
+                }
+                ]
+              }
+            };
+            node.status({fill:"blue",shape:"ring",text:"Uploading document"});
+            concept_insights.corpora.createDocument(params, function(err,res) {
+              node.status({fill:"blue",shape:"ring",text:"Document uploaded. Started indexing"});
+              if (err) {
+                node.status({});
+                var message = "Creating document error: "+err.error;
+                return node.error(message, msg);
+              } else {
+                if (cb) cb(); 
+              }         
+            });
+          } else {
+            var message = "Creating document error: Missing document name, label or content type";
+            return node.error(message, msg);
+          }
+        }          
+
+        function checkStatus(fileLength) {
+          //Calculate interval based on length of file, default to 10 seconds
+          var interval = 10000;
+          if (fileLength > 0) interval = (2*fileLength) + 5;
+          
+          //Loop and check the status of the document 
+          var statusInterval = setInterval(function(){
+            concept_insights.corpora.getDocumentProcessingState(params, function(err,res) {
+              if (err) {
+                node.status({});
+                var message = "Document processing state error: "+err.error;
+                clearInterval(statusInterval);
+                return node.error(message, msg);
+              } else {
+                //get status
+                if (res.status == 'ready') {
+                  node.status({fill:"green",shape:"ring",text:"Document ready"});
+                  return clearInterval(statusInterval);
+                }
+              }
+            });
+          },interval);
+        }
+      });
+    });
+  }
+
+  RED.nodes.registerType('watson-concept-insights-upload-document', uploadDocumentNode, {
+    credentials: {
+      username: {type:"text"},
+      password: {type:"password"}
+    }
+  });
+
+  function setupConceptInsightsNode(msg,config,node,callback) {
+  
+    //Check for payload
+    if (!msg.payload) {
+      var message = 'Missing property: msg.payload';
+      node.error(message, msg)
+      return;
+    }
+
+    //Check credentials
+    username = username || node.credentials.username;
+    password = password || node.credentials.password;
+
+    if (!username || !password) {
+      var message = 'Missing Concept Insights service credentials';
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Variable 'message' is already declared in line 376.](https://www.codacy.com/app/rezgui-y/node-red-node-watson/file/2392377031/issues/source?bid=3225115&fileBranchId=3228814#l386)